_plugins/locale_url_filter.rb

```diff
@@ -0,0 +1,17 @@
+module Jekyll
+  module LocaleUrlFilter
+    LOCALE_STRING_CONSTANT = '__LOCALE_BASE_URL__'.freeze
+
+    def locale_url(path = '')
+      site_base_url = @context.registers[:site].baseurl
+      locale_prefix = @context.registers[:page]['base_path']
+      site_base_url.to_s + locale_prefix.to_s + path
```

Comment:
If we did something like this, we wouldn't have to worry that each part has a slash in the right place, this would take care of that for us?

```suggestion
      File.join(site_base_url.to_s, locale_prefix.to_s, path)
```

Reply:
> If we did something like this, we wouldn't have to worry that each part has a slash in the right place, this would take care of that for us?

I was reading this blog post earlier, which had a good point I'd not have thought of myself as to why we might not want to do this:

https://www.honeybadger.io/blog/why-is-uri-join-so-counterintuitive/

>**EDIT 8/12/2016:** After publishing this article I received a couple of tweets suggesting I use `File.join` for this purpose. This has the benefit of avoiding double slashes, ie. `/my//path` but will break on OSs like Windows, where the path separator isn't a forward-slash.

Reply:
Yeah that's why I picked `File.join` ... and felt like its a safe assumption we're on unix/linux systems for builds for the most part, but 🤷

Reply:
Yeah, that feels like it'd be fine in most cases, but doesn't leave me feeling great for the possibility of someone running Windows.

We could do it manually?

```rb
[site_base_url, locale_prefix, path].map { |p| String(p).gsub(%r{^/|/$}, '') }.join('/')
```

Or use `File.join` and coerce the separator to the one we want?

```rb
File.join(site_base_url.to_s, locale_prefix.to_s, path).gsub(File::SEPARATOR, '/')
```

Reply:
Looking at the docs (and confirming in the source), it appears `File.join` always uses a forward slash regardless of platform:

https://ruby-doc.org/core-2.6.2/File.html#method-c-join
https://github.com/ruby/ruby/blob/4337da377e5d2fcebf0da9600f6e09727ea9a624/file.c#L5013

Reply:
Hm! Good find. Seems like a good fit then.